lib/chef/knife/vsphere_vm_nic_delete.rb

```diff
@@ -0,0 +1,48 @@
+#
+# Author:: Scott Williams (<scott@backups.net.au>)
+# License:: Apache License, Version 2.0
+#
+require 'chef/knife'
+require 'chef/knife/base_vsphere_command'
+require 'rbvmomi'
+require 'netaddr'
+
+class Chef::Knife::VsphereVmNicDelete < Chef::Knife::BaseVsphereCommand
+  banner 'knife vsphere vm nic delete VMNAME NICNAME'
+
+  common_options
+
+  def run
+    $stdout.sync = true
+
+    vmname = @name_args[0]
+    if vmname.nil?
+      show_usage
+      fatal_exit('You must specify a virtual machine name')
+    end
+
+    nicname = @name_args[1]
+    if nicname.nil?
+      show_usage
+      fatal_exit('You must specify the name of the NIC to delete')
+    end
+
+    vim_connection
+    vm = get_vm(vmname) || abort('VM not found')
+
+    vm.config.hardware.device.each.grep(RbVmomi::VIM::VirtualEthernetCard) do |a|
```

Comment:
Good idea. 
Also this command requires the VM to be powered off, should we check the VM state before the delete?